src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SingleExtensionEvalFunction.java

```diff
@@ -347,6 +401,32 @@ private void validateAllImportsAreGenerated(
     }
   }
 
+  /**
+   * @return usages with all information removed that does not influence the evaluation of the
+   *     extension, only the validation against imports that comes afterward.
```

Comment:
IMO remove the "only the validation ..." part, since locations don't even matter for that. And the comments inside the method explain the removals well enough.